boot/bootutil/src/loader.c

```diff
@@ -703,14 +712,32 @@ boot_validated_swap_type(struct boot_loader_state *state,
     swap_type = boot_swap_type_multi(BOOT_CURR_IMG(state));
     if (BOOT_IS_UPGRADE(swap_type)) {
         /* Boot loader wants to switch to the secondary slot.
-         * Ensure image is valid.
+         * Ensure image is valid. This checks header, hash/signature and trailer
          */
         rc = boot_validate_slot(state, BOOT_SECONDARY_SLOT, bs);
+	/* Check for network update type */
         if (rc == 1) {
             swap_type = BOOT_SWAP_TYPE_NONE;
         } else if (rc != 0) {
             swap_type = BOOT_SWAP_TYPE_FAIL;
         }
+
+	vtable_addr = (uint32_t)hdr + hdr->ih_hdr_size;
+	vtable = (uint32_t *)(vtable_addr);
+	reset_addr = vtable[1];
+	if (reset_addr > PM_nrf5340pdk_nrf5340_cpunet_B0N_ADDRESS) {
+		uint32_t fw_size = hdr->ih_img_size;
+
+		BOOT_LOG_INF("Starting netowrk core update");
+		rc = do_network_core_update(NULL, vtable, fw_size);
+		if (rc != 0) {
+			swap_type = BOOT_SWAP_TYPE_FAIL;
+		} else {
+			BOOT_LOG_INF("Done updating network core");
+			BOOT_SWAP_TYPE(state) = swap_type = BOOT_SWAP_TYPE_NONE;
+			rc = swap_erase_trailer_sectors(state, secondary_fa);
```

Comment:
Maybe find a better way than just deleting the whole trailer. As we only need to remove the flag which initiates the swap so the SWAP_TEST type.